arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/continuations/EagerEffectScope.kt

```diff
@@ -186,6 +188,38 @@ public interface EagerEffectScope<in R> {
    */
   public suspend fun ensure(condition: Boolean, shift: () -> R): Unit =
     if (condition) Unit else shift(shift())
+
+  /**
+   * Initializes an "attempt block", which encloses an action for which
+   * you want to catch any possible `shift`ed outcome. This should be used
+   * in combination with `catch`.
+   *
+   * ```
+   * attempt { ... } catch { ... }
+   * ```
+   *
+   * The [effect] may `shift` into a different `ErrorScope`, giving
+   * the change to a later `recover` to change the shifted value.
+   * This is useful to simulate re-throwing of exceptions.
+   */
+  @OptIn(ExperimentalTypeInference::class)
+  public suspend fun <E, A> attempt(
```

Comment:
In the Kotlin Stdlib functions build using the BuilderInference would start with `build*` e.g.: `buildList`, `buildString`, etc.

So here we could name this:
`buildEffect` and `buildEagerEffect`
wdyt cc @nomisRev, @serras

Reply:
I think that's inaccurate, it's only true for some APIs. Makes sense that `buildXXX` DSLs benefit from `@BuilderInference`, but it's also heavily used in KotlinX Coroutines wherever generic DSLs are being used.

https://github.com/Kotlin/kotlinx.coroutines/search?q=BuilderInference